### PR TITLE
Fuzzy search

### PR DESCRIPTION
**Pull request by mellogarrett**
- Use [fuse.js](https://github.com/krisk/Fuse) for fuzzy search
- Fix `<Linkify>` issues with React keys

_unrelated to title, but big change that was blocking me from creating this PR_
- Replace Husky pre-commit hooks with `"test": "npm-run-all test:*"` which can be seen in [other projects](https://github.com/LambdaSchoolEngineering/atlas-db/blob/17514eee8da1ffbd4be82e033cc94ee677cab934/package.json#L46-L49). Husky kept complaining that it couldn't find my version of Node and wouldn't let me commit. Seems to be a bug with Husky & nvm integration. No idea why it's just now coming up... Anyway, as long as we run `npm run test` before we push commits, everything will be fine and dandy. If we don't, Travis will yell at us.

### CODE REVIEW

**Comment by boblauer** [comment]
Looks like Travis threw up, not sure what's going on there.

File: components/dropdown-question/index.tsx [hunk 1]
```diff
@@ -42,10 +45,7 @@ export const DropdownQuestionComponent = ({ question, onChange, style, className
         selectedItem,
         clearSelection,
       }) => {
-        const suggestions = choices.filter((choice) => {
-          const lowercaseDisplayValue = (choice.display || choice.value).toString().toLowerCase();
-          return inputValue === null || lowercaseDisplayValue.includes(inputValue.toLowerCase());
-        });
+        const suggestions = inputValue ? fuse.search(inputValue) : choices;
```
Comment (MarkDKoslow) [inline components/dropdown-question/index.tsx]: Wow O____o crazy how simple that is

**Comment by mellogarrett** [comment]
Performance is okay. There’s a slight lag opening the list.

**Comment by boblauer** [comment]
Is the lag from the fuzzy searching itself, or perhaps a rendering issue?

**Comment by mellogarrett** [comment]
Not sure to be honest. Very likely it could be a rendering issue. It's not bad at all, just noticeable.

**Pull request merged**